PocketKit/Sources/PocketKit/Article/Highlights/ArticleComponentHighlights.swift

```diff
@@ -146,26 +158,34 @@ extension Array where Element == ArticleComponent {
         return mergedComponents
     }
 
-    private func normalizedComponents(_ patchedBlob: String) throws -> [String] {
-        var patchedComponents = patchedBlob.components(separatedBy: HighlightConstants.componentSeparator)
+    /// Extract text components, identified by the separator tag, from a blob.
+    /// Highlights are delimited by the corresponding start and end tag.
+    /// If an highlight spans more than one component, it gets broken up
+    /// into multiple highlights, one per component.
+    /// - Parameter text: the text blob that contains separator tags and highlights.
+    /// - Returns: The array of text components with highlights.
+    private func textComponentsWithHighlights(_ text: String) throws -> [String] {
+        var patchedComponents = text.components(separatedBy: HighlightConstants.componentSeparator)
 
-        let scanner = Scanner(string: patchedBlob)
+        let scanner = Scanner(string: text)
         var componentCursor = 0
         var tagStack = [String]()
 
         while !scanner.isAtEnd {
             guard scanner.scanUpToString(HighlightConstants.commonTag) != nil else {
                 return patchedComponents
             }
-            let beforeIndex = Swift.max(patchedBlob.index(before: scanner.currentIndex), patchedBlob.startIndex)
-            let character = String(patchedBlob[beforeIndex])
+            let beforeIndex = Swift.max(text.index(before: scanner.currentIndex), text.startIndex)
+            let character = String(text[beforeIndex])
             if character == HighlightConstants.startTagIdentifier {
                 tagStack.append(HighlightConstants.highlightStartTag)
             }
             if character == HighlightConstants.endTagIdentifier {
                 guard !tagStack.isEmpty else {
                     // in the entire blob, we are not supposed to find a closing tag without an opening tag
-                    throw HighlightError.invalidPatch(componentCursor)
+                    // if we do, it's probably an error from the Diff Match Patch: continue but log the error
+                    Log.capture(message: "Error parsing highlights at index \(componentCursor)")
```

Comment:
This will log to Sentry. I dont think we should use Sentry as a logging platform. It should be used for things we can/and will take action on. Switching to .info will change this to a log message, i believe.